massgov/mayflower
# DP-25175 headings as targets for TOC links

Any PRs being created needs a changelog.txt file before being merged into dev. See: [Change Log Instructions](https://github.com/massgov/mayflower/blob/develop/docs/for-developers/changelog-instructions.md)


## Description


- Move the link target (ID) from `span` to `h2.ma__comp-heading` to enable screen reader announce the heading when a TOC link lands on its target.
- Remove an empty `span` used to contained the link target.
- Replace `div`s with unordered list for the TOC markup to improve semantics
- Remove `aria-labelledby` from nav to prevent “TABLE OF CONTENTS” to be announced repeatedly by screen readers.
- Remove the menu expand states in the button labels in mobile to simplify the labels.
- Add aria roles to the TOC component.

No visual change with this PR.

## Related Issue / Ticket

- [JIRA issue](https://massgov.atlassian.net/browse/DP-25175)
- [Github issue]()

## Steps to Test


1. Go to any page with TOC. e.g. /patterns/05-pages-guide/05-pages-guide.html
2. In markup, ind `aria-labelledby` is removed from `nav.ma__sticky-toc__links`.
3. Check the markup for the TOC.  In `nav.ma__sticky-toc__links`, TOC links are structured as an unordered link.  aria-roles are added.

      
         <ul role="menu">
                <li class="ma__sticky-toc__link" data-link="#buying-or-renting-a-home" role="none">
                     <a href="#buying-or-renting-a-home" role="menuitem">
                           <svg xmlns="http://www.w3.org/2000/svg" aria-hidden="true" width="35" height="35" viewBox="0 0 35 35">...</svg>Buying or Renting a Home</a>
                </li>
                ...   
         </ul>
      

4. Click one of TOC links to go to its target section.  Find its functionality hasn't changed and the section title doesn't get covered by the sticky TOC.
5. Check the markup for a section heading.  In `div.ma__stacked-row__title`, no `span`, `h2.ma__comp-heading` has the TOC link target ID and `tabindex="-1".
      ``` 
         <div class="ma__stacked-row__title">
            <h2 class="ma__comp-heading sticky-toc-jump-target" id="buying-or-renting-a-home" tabindex="-1">Buying or Renting a Home</h2>        
         </div>
      ```


## Screenshots
Use something like [licecap](http://www.cockos.com/licecap/) to capture gifs to demonstrate behaviors.


## Additional Notes:

Anything else to add?

#### Impacted Areas in Application


*

#### @TODO


*

#### Today I learned...
